src/Data/ByteString/Base16.hs

```diff
@@ -139,6 +140,21 @@ isBase16 :: ByteString -> Bool
 isBase16 bs = isValidBase16 bs && isRight (decodeBase16Untyped bs)
 {-# INLINE isBase16 #-}
 
+-- | Try to parse something that's assumed to be base16 encoded into the proper type
+--
+-- === __Examples__:
+--
+-- >>> parseBase16 "666f6"
+-- Left "invalid bytestring size"
+--
+-- >>> parseBase16 "666f"
+-- Right "666f"
+--
+parseBase16 :: ByteString -> Either Text (Base16 ByteString)
+parseBase16 bs
+  | isValidBase16 bs = assertBase16 bs <$ decodeBase16Untyped bs
+  | otherwise = Left "no valid Base16 format"
```

Comment:
```suggestion
parseBase16 bs = assertBase16 bs <$ decodeBase16Untyped bs
```

Reply:
The decode is the test, and I'll migrate away from the `is*` functions in a major version release.